plugins-bitcoincom/cordova-plugin-qrreader/src/ios/QRReader.swift

```diff
@@ -0,0 +1,196 @@
+        captureSession = AVCaptureSession()
+        
+        guard let videoCaptureDevice = AVCaptureDevice.defaultDevice(withMediaType: AVMediaTypeVideo) else {
+            // TODO: Handle the case without permission "Permission"
+            onFailed(QRReaderError.ERROR_PERMISSION_DENIED)
+            return
+        }
+        
+        let videoInput: AVCaptureDeviceInput
+        
+        do {
+            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
+        } catch {
+            // TODO: Handle this case "Retry"
+            onFailed(QRReaderError.ERROR_PERMISSION_DENIED)
+            return
+        }
+        
+        guard captureSession.canAddInput(videoInput) else {
+            onFailed(QRReaderError.ERROR_SCANNING_UNSUPPORTED)
+            return
+        }
+        captureSession.addInput(videoInput)
+        
+        let metadataOutput = AVCaptureMetadataOutput()
+        
+        guard captureSession.canAddOutput(metadataOutput) else {
+            onFailed(QRReaderError.ERROR_SCANNING_UNSUPPORTED)
+            return
+        }
+        captureSession.addOutput(metadataOutput)
+        
+        metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+        metadataOutput.metadataObjectTypes = [AVMetadataObjectTypeQRCode]
+        
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = cameraView.layer.bounds
+        previewLayer.videoGravity = AVLayerVideoGravityResizeAspectFill
+        
+        cameraView.layer.addSublayer(previewLayer)
+        superView.insertSubview(cameraView, belowSubview: webView)
+        
+        captureSession.startRunning()
+    }
+    
+    func stopReading(_ command: CDVInvokedUrlCommand){
+        captureSession.stopRunning()
+    }
+}
+
+// Private methods
+//
+extension QRReader {
+    
+    fileprivate func callback(_ command: CDVInvokedUrlCommand, status: CDVCommandStatus) {
+        callback(command, status: status, message: nil)
+    }
+    
+    fileprivate func callback(_ command: CDVInvokedUrlCommand, status: CDVCommandStatus, message: String?) {
+        guard let callbackId = command.callbackId
+            , let commandDelegate = self.commandDelegate else {
+                return
+        }
+        
+        var pluginResult: CDVPluginResult
+        
+        // Callback
+        if let _ = message  {
+            pluginResult = CDVPluginResult(status: status, messageAs: message)
+        } else {
+            pluginResult = CDVPluginResult(status: status)
+        }
+        
+        commandDelegate.send(pluginResult, callbackId: callbackId)
+    }
+    
+    func onSuccess(_ result: String) {
+        guard let readingCommand = self.readingCommand else {
+            return
+        }
+        
+        callback(readingCommand, status: CDVCommandStatus_OK, message: result)
+        self.readingCommand = nil
+    }
+    
+    func onFailed(_ error: QRReaderError) {
```

Comment:
This function should be `fileprivate` like the others. How about making the whole extension fileprivate instead?

Reply:
Yes, sure, go ahead and fix it.

Reply:
Move to a private repo, will be fixed there.